tests/test_core.py

```diff
@@ -7179,6 +7179,235 @@ def test_mallinfo(self):
+print("m2 read");
+m2.ccall('myread0','number',[],[]);
+
+print("m0 read embed");
+m0.ccall('myreade','number',[],[]);
+print("m1 read embed");
+m1.ccall('myreade','number',[],[]);
+print("m2 read embed");
+m2.ccall('myreade','number',[],[]);
+
+section = "parent m0 writes and reads children's files.";
+print("m0 write m1");console.log("");
+m0.ccall('mywrite1','number',[],[]);
+print("m0 read m1");
+m0.ccall('myread1','number',[],[]);
+print("m0 write m2");console.log("");
+m0.ccall('mywrite2','number',[],[]);
+print("m0 read m2");
+m0.ccall('myread2','number',[],[]);
+print("m1 read");
+m1.ccall('myread0','number',[],[]);
+print("m2 read");
+m2.ccall('myread0','number',[],[]);
+print("m0 read m0");
+m0.ccall('myread0','number',[],[]);
+''')
+
+    open('proxyfs_pre.js', 'w').write(r'''
+if (typeof Module === 'undefined') Module = {};
+Module["noInitialRun"]=true;
+Module["noExitRuntime"]=true;
+''')
+
+    open('proxyfs_embed.txt', 'w').write(r'''test
+''')
+
+    open('proxyfs_test.c', 'w').write(r'''
+#include <stdio.h>
+
+int
+mywrite1(){
+  FILE* out = fopen("/working/hoge.txt","w");
+  fprintf(out,"test1\n");
+  fclose(out);
+  return 0;
+}
+
+int
+myread1(){
+  FILE* in = fopen("/working/hoge.txt","r");
+  char buf[1024];
+  int len;
+  if(in==NULL)
+    printf("open failed\n");
+
+  while(! feof(in)){
+    if(fgets(buf,sizeof(buf),in)==buf){
+      printf("%s",buf);
+    }
+  }
+  fclose(in);
+  return 0;
+}
+int
+mywrite2(){
+  FILE* out = fopen("/working2/hoge.txt","w");
+  fprintf(out,"test2\n");
+  fclose(out);
+  return 0;
+}
+
+int
+myread2(){
+  {
+    FILE* in = fopen("/working2/hoge.txt","r");
+    char buf[1024];
+    int len;
+    if(in==NULL)
+      printf("open failed\n");
+
+    while(! feof(in)){
+      if(fgets(buf,sizeof(buf),in)==buf){
+        printf("%s",buf);
+      }
+    }
+    fclose(in);
+  }
+  return 0;
+}
+
+int
+mywrite0(int i){
+  FILE* out = fopen("hoge.txt","w");
+  fprintf(out,"test0_%d\n",i);
+  fclose(out);
+  return 0;
+}
+
+int
+myread0(){
+  {
+    FILE* in = fopen("hoge.txt","r");
+    char buf[1024];
+    int len;
+    if(in==NULL)
+      printf("open failed\n");
+
+    while(! feof(in)){
+      if(fgets(buf,sizeof(buf),in)==buf){
+        printf("%s",buf);
+      }
+    }
+    fclose(in);
+  }
+  return 0;
+}
+
+int
+myreade(){
+  {
+    FILE* in = fopen("proxyfs_embed.txt","r");
+    char buf[1024];
+    int len;
+    if(in==NULL)
+      printf("open failed\n");
+
+    while(! feof(in)){
+      if(fgets(buf,sizeof(buf),in)==buf){
+        printf("%s",buf);
+      }
+    }
+    fclose(in);
+  }
+  return 0;
+}
+''')
+
+    Popen([PYTHON, EMCC,
+           '-o', 'proxyfs_test.js', 'proxyfs_test.c',
+           '--embed-file', 'proxyfs_embed.txt', '--pre-js', 'proxyfs_pre.js',
+           '-s', 'MAIN_MODULE=1']).communicate()
+    Popen(['cp', 'proxyfs_test.js', 'proxyfs_test1.js']).communicate()
+    Popen(['cp', 'proxyfs_test.js', 'proxyfs_test2.js']).communicate()
```

Comment:
i feel like i'm missing something here. if the js files are all just copies of each other, how do they do different things?